proxy/proxy.go

```diff
@@ -128,7 +147,26 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 				return
 			}
 			if !ok {
-				p.handlePaymentRequired(w, r, target.Name, target.Price)
+				price, err := target.pricer.GetPrice(
+					r.Context(), r.URL.Path,
+				)
+				if err != nil {
+					prefixLog.Errorf("error getting"+
+						" resource price: %v", err)
```

Comment:
We need to send a response here with `sendDirectResponse()`, probably an internal server error would be appropriate here.  Same in the `case` above.